### PR TITLE
Flexible calibration

### PR DESCRIPTION
**Pull request by sfalexrog**
This is a major overhaul of the calibrator software. Added features:

* configurable calibrator and ArUco detector parameters: now pretty much every parameter can be configured;
* custom YAML writer: The calibration file should be completely compatible with `camera_info_manager`;
* file-based calibration: You can now specify a folder with images to perform non-interactive calibration.

Improvements:

* `board` topic is now being published correctly;
* images are now filtered based on the amount of found board features (this should make the calibrator more robust);
* saved images are "raw" images, as seen in their topics;
* code is more structured now, with function/method documentation, split across several files;
* OpenCV 3.2 compatibility.

This should make further additions to the package a lot simplier.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,52 +1,113 @@
 # charuco_calibration
+
 Camera calibration via charuco board ROS package.
-> If you want to calibrate the camera on the embedded computer (like the Raspberry Pi), you should install this package on your laptop and use remote calibration launch file. 
+> If you want to calibrate the camera on the embedded computer (like the Raspberry Pi), you should install this package on your laptop and use remote calibration launch file.
+
+Currently the package provides the following nodes:
+
+* `charuco_calibration_node` - an interactive ChArUco calibration node that uses an image topic to get calibration images;
```
Comment (okalachev) [inline README.md:8]: Why not to merge it to one node?

Comment (sfalexrog) [reply README.md:8]: I'd prefer the nodes to be simplier, with less code paths, less code overall. Besides, right now the two nodes have significantly different behaviour: one of them expects an image topic and user input, the other just looks at images in a folder and creates a calibration file.

**Comment by okalachev** [comment]
Is it gonna be usable in context of web camera calibration? Now looks a lot complicated.

**Comment by sfalexrog** [comment]
@okalachev, what's the problem with webcam calibration? You can use any source that will eventually post the image as `sensor_msgs/Image`. Even better, with file based calibration you can run the interactive node, take a bunch of pictures without knowing exact board dimensions, then measure the board and run the calibration again with proper board parameters.

**Comment by okalachev** [comment]
No, I'm talking about web GUI based camera calibration.

**Comment by sfalexrog** [comment]
@okalachev, well, at the very least this provides a good starting point for any calibration tool, web-based or not. I believe I can create a proof of concept tool in a couple of days, if this is merged.

**Comment by okalachev** [comment]
Who should review this, @goldarte?

**Comment by sfalexrog** [comment]
Just in case, I've started working on a service-based version that can be used as a backend for a web calibration tool: https://github.com/sfalexrog/charuco_calibration/tree/web_calibration_poc. I'm still concerned about running the whole thing on a Raspberry, though: 50+ images typically take quite a lot of time to process even on a powerful laptop.

**Pull request merged**